src/agent/onefuzz/src/syncdir.rs

```diff
@@ -231,11 +239,19 @@ impl SyncedDir {
                 let file_name = item
                     .file_name()
                     .ok_or_else(|| anyhow!("invalid file path"))?;
-                if ignore_dotfiles && file_name.to_string_lossy().starts_with('.') {
+                let file_name_str = file_name.to_string_lossy();
+
+                // explicitly ignore azcopy temporary files
+                // https://github.com/Azure/azure-storage-azcopy/blob/main/ste/xfer-remoteToLocal-file.go#L35
+                if file_name_str.starts_with(".azDownload-") {
                     continue;
                 }
-                event!(event.clone(); EventData::Path = item.display().to_string());
 
+                if ignore_dotfiles && file_name_str.starts_with('.') {
```

Comment:
It looks like we've been exactly duplicating logic in each branch that should always be the same. Since it is becoming a bit more complex now, consider introducing a helper function like `should_ignore_file(path, ignore_dotfiles) -> bool` that will tell us whether to `continue` based on filename.